### DISCUSSION TITLE
Issue setting up my orgs custom Demo Object

### DISCUSSION THREAD

**Discussion by bneighborshcp** [deejay-hub/timeline-lwc#223] [Q&A] [answered]
I'm trying to set up the Time Warp on my Account. I think I have my MetaData set up incorrectly but other members of my team do not know how to fix it. 
![image](https://user-images.githubusercontent.com/115509182/194962271-16dbbd80-b9f1-4259-98a4-b376c0b15087.png)
This is the Demo Relations to the Opp
![image](https://user-images.githubusercontent.com/115509182/194962647-3e119028-508f-41d3-9742-93946fafd9e0.png)
I will be repeating this on the Case as well. This will provide my teams with visibility on all the things happening on the account and what has happened under each related Opp/Case/Demo.

**Comment by deejay-hub** [comment]
Okay. Let's walkthrough the setup step by step. That might help you in the future to map other objects.

Firstly...for anyone reading this you should check out the configuration guide [here](https://salesforce.quip.com/ahFCA9VBEKtr).  This goes through step by step mapping standard, custom and junction objects with samples.

Given you've taken the time to raise a discussion I will walk you through it for your specific example.

It sounds like you have a custom object called Demo__c that you are linking to the standard Opportunity object via a lookup relationship.

Step 1 -  Identify the relationship name that defines how the parent and child Objects are linked. In your case you created a custom relationship called Current Demo Record (Current_Demo_Record__c)
<img width="888" alt="image" src="https://user-images.githubusercontent.com/7021883/194963597-39e1d2b5-7202-47d6-8b80-ba78979ac71b.png">

Note down the value for the Child Relationship. In your case you called it *Opportunities*. This is the value we need for the Relationship Name. Note that since it's a custom field (not a standard one) we need to append __r. So the relationship name Salesforce assigns to this is called Opportunities__r.

Step 2. Decide where you want to plot Demo records in the timeline. This is typically a field on the child object itself (Demo__c). Looks like you wanted the Created Date which maps to the API name CreatedDate.

Step 3: Decide which field value to appear when we plot on the timeline. Looks like you want the value in the field Type2__c to be shown. Again this is typically a field on the child object (Demo__c).

Step 4: Create a new metadata type record with the values from above.

NOTE THAT YOU HAVE A TYPO IN THE RELATIONSHIP NAME FIELD AND YOU WERE USING OPPORTUNITY AND NOT OPPORTUNITIES.
<img width="1150" alt="image" src="https://user-images.githubusercontent.com/7021883/194964546-0302e4cf-3a12-4d76-858c-27e10f098583.png">

**Reply by bneighborshcp** [reply]
Thank you DJ. Is my issue that I'm trying to set it up on the Account to get the child of the Opp that is on the account?

**Reply by deejay-hub** [reply]
If that's your expectation then yes that's the issue. We execute a SOQL query like this SELECT Id, 'Field', (SELECT Id, Type2__c FROM Opportunities__r) FROM Opportunity WHERE Id = 'Opty Id'. Another way to think about it is that Time Warp shows you records available in related lists. If you can't configure a related list to show an object from the parent then Time Warp can't plot it. So for Account if you can't have a standard related list for Demo's then Time Warp can't plot it.

If you don't have a lookup or master detail to the object then you don't have a relationship to query across.

**Comment by deejay-hub** [comment, marked as answer]
Also you mention that you are trying to set this up on Account. Note that the above is the setup for Opportunity. For Account you'd have to repeat a very similar process. The Parent Object in the metadata setup should be Account. But with the above instructions it should make more sense.

**Reply by bneighborshcp** [reply]
Am I not able to see the Grandchild objects?

**Reply by deejay-hub** [reply]
No. Grandchild objects are **not** supported. This would require a recursive query which wouldn't scale and create performance issues. There is a pull request to offer the ability to create an Apex class that we are looking to introduce in a future version that could meet the needs to query beyond direct child relationships.

The only exception is the Account to Task/Event/Email and Call. Indirect relationships are supported here by virtue of the fact that Salesforce will populate Open Activities and Activity History objects with grandchild records.

**Reply by bneighborshcp** [reply]
Thank you for the information. I will have to adjust plans accordingly.
